src/map.cpp

```diff
@@ -1304,7 +1305,14 @@ bool map::can_move_furniture( const tripoint &pos, player *p )
     }
 
     ///\EFFECT_STR determines what furniture the player can move
-    if( p != nullptr && p->str_cur < required_str ) {
+    int adjusted_str = p->str_cur;
+    if( p->has_effect( effect_riding ) && p->mounted_creature ) {
+        auto mons = p->mounted_creature.get();
+        if( mons->has_flag( MF_RIDEABLE_MECH ) && mons->type->mech_str_bonus != 0 ) {
+            adjusted_str = mons->type->mech_str_bonus;
+        }
+    }
+    if( p != nullptr && adjusted_str < required_str ) {
```

Comment:
That check for `p` being non-null is to late. You have already dereferenced it above, causing undefined behavior when `p` is actually `null`.